### ISSUE TITLE
Hyperpixel background goes green and touch controller stops working when starting Octoprint

### ISSUE THREAD

**Issue by dmberrycom** [pimoroni/hyperpixel4#83]
**Describe the bug**

Screen boots okay and shows Raspbian boot messages as normal and then launches into the OctoDash UI normally with the touch controller working. When octoprint starts the screens background which is normally black goes green and the touch input stops working.

**To Reproduce**

Start Octoprint 
Will happens shortly after boot if the service is enabled on boot, and only when the service is started manually if disabled on boot.

What install steps did you go through so far? Did you install from GitHub or use our one-line-installer?

Tried the basic one line installer and also manually using the pi4-i2c-fix branch

**Your HyperPixel 4**

Rectangular / Touch (Purchased July 2020)

Connection method to Pi does not affect the issue, tested either connected directly using the connector in the package, a short extension GPIO extension cable or a HAT breakout.

Running on a Pi 4 / 4GB

Initially tried with the running a version of Raspbian that is running a unifi controller and octoprint for a while, but also tried with fresh installs of Raspbian Lite and Full with only installing octoprint and hyperpixel

1. What model of Raspberry Pi are you using (the result of `cat /proc/cpuinfo | grep Revision` can tell us this exactly
Revision        : b03112

2. Which OS release are you using: `lsb_release --description` and `uname -r`
 Raspbian GNU/Linux 10 (buster) 4.19.118-v7l+

3. Is this a fresh OS setup, or one you might have previously installed scripts/add-ons on?
Tried both Fresh & Existing with other programs

**Extra debugging information**

* `dmesg | grep Goodix` 
Note these are boot time messages, after boot screen is working okay, when the issue occurs no other entries are present.

[    4.199603] Goodix-TS 7-005d: i2c test failed attempt 1: -6
[    4.238709] Goodix-TS 7-005d: i2c test failed attempt 2: -6
[    4.278316] Goodix-TS 7-005d: I2C communication failure: -6
[    4.280313] Goodix-TS 7-0014: ID 911, version: 1060
[    4.301579] input: Goodix Capacitive TouchScreen as /devices/platform/i2c@0/i2c-7/7-0014/input/input0

*i2c Output

ls /dev/i2c-*
/dev/i2c-7

pi@raspberrypi:~ $ i2cdetect -y 7
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- UU -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --

**Comment by dmberrycom** [comment]
Resolved, 

Issue was with the enclosure plugin in octoprint, after doing a clean install and not importing the backup from octoprint, the screen worked correctly.

This indicatied it was something in the configuration of an Octoprint plugin. I traced it down to the enclosure plugin and it was fairly obvious then, I had been testing i2C before purchasing the Hyperpixel and the enclosure plugin had been configured to talk to outputs over i2C for power control. After changing those output to a script rather than natively trying to touch i2c this issue stopped.

**Comment by dmberrycom** [comment]
Resolved,

Issue was with the enclosure plugin in octoprint, after doing a clean install and not importing the backup from octoprint, the screen worked correctly.

This indicatied it was something in the configuration of an Octoprint plugin. I traced it down to the enclosure plugin and it was fairly obvious then, I had been testing i2C before purchasing the Hyperpixel and the enclosure plugin had been configured to talk to outputs over i2C for power control. After changing those output to a script rather than natively trying to touch i2c this issue stopped.
